02/fuzzygwalchmei/game.py

```diff
@@ -0,0 +1,81 @@
+#!python3
+# Code Challenge 02 - Word Values Part II - a simple game
+# http://pybit.es/codechallenge02.html
+
+import itertools
+import random
+
+from data import DICTIONARY, LETTER_SCORES, POUCH
+
+NUM_LETTERS = 7
+
+
+def draw_letters():
+    """Pick NUM_LETTERS letters randomly. Hint: use stdlib random"""
+    return random.sample(POUCH, NUM_LETTERS) 
+
+
+def input_word(draw):
+    """Ask player for a word and validate against draw.
+    Use _validation(word, draw) helper."""
+    word = input(f'What word did you want to submit?\nYour current letters are {draw}: ')
+    return _validation(word, draw)
+
+
+
+def _validation(word, draw):
+    """Validations: 1) only use letters of draw, 2) valid dictionary word"""
+    if not word in get_possible_dict_words(list(draw)):
+        raise ValueError
+
+
+# From challenge 01:
+def calc_word_value(word):
+    """Calc a given word value based on Scrabble LETTER_SCORES mapping"""
+    return sum(LETTER_SCORES.get(char.upper(), 0) for char in word)
+
+
+# Below 2 functions pass through the same 'draw' argument (smell?).
+# Maybe you want to abstract this into a class?
+# get_possible_dict_words and _get_permutations_draw would be instance methods.
+# 'draw' would be set in the class constructor (__init__).
+def get_possible_dict_words(draw):
+    """Get all possible words from draw which are valid dictionary words.
+    Use the _get_permutations_draw helper and DICTIONARY constant"""
+    return [word for word in _get_permutations_draw(draw) if word.lower() in DICTIONARY]
```

Comment:
seems we do a `word in`, with a list it scans all items, would a `set` be better?